### ISSUE TITLE
Epic: evidence-governed workflow improvement for Golem

### ISSUE THREAD

**Issue by kstruzzieri** [kstruzzieri/go-llm#487]
Build a local, manual loop that ties human feedback to an exact Golem run, turns selected examples into replayable tests, compares a small workflow-instruction change against the old version, and produces a patch that a human must review and apply.

This is not model training. The model does not rewrite itself. The thing that improves is a version-controlled workflow playbook, and only after evidence and human review.

**Initial workflow:** code review only.
**Initial surface:** interactive Golem REPL only.
**Likely components:** `feedback`, Golem REPL, `projectcontext`, `agenttrace`, `llm-bench`; existing AgentFlow controls remain an optional, separate apply boundary.

## Epic outcome

As a Golem user, I can mark an answer useful or not useful, preserve enough local provenance to reproduce it, deliberately promote suitable examples into an evaluation corpus, and review a small workflow-playbook patch that has been tested against held-out cases.

## Epic acceptance criteria

- [ ] A user can rate the last eligible Golem answer as useful or not useful, with an optional bounded reason.
- [ ] The content-light receipt is linked to an opaque run ID and session ID without storing prompt or answer content. A separate sensitive rating row contains only the verdict and the reason the user explicitly supplies.
- [ ] Existing implicit retrieval-ranking behavior is unchanged.
- [ ] A content-full trace remains separately opt-in; feedback without a trace is retained but is clearly ineligible for replay export.
- [ ] An eligible trace can be converted into a valid `llm-bench` case with an exact playbook fingerprint and resolvable tool-schema snapshot.
- [ ] A human, not a model, decides which feedback cases enter development and held-out evaluation sets and supplies the expected behavior.
- [ ] Baseline and candidate playbooks are evaluated on the same held-out cases, model route, and tool contract.
- [ ] Promotion fails closed on missing/stale evidence, incomplete pairs, quality regression outside the approved margin, or restraint regression.
- [ ] The improver emits a proposed diff and evidence manifest only. It cannot change the fixed core prompt, safety rules, tool approvals, model weights, memory, source branch, or active playbook automatically.
- [ ] A human reviews and applies any accepted change.
- [ ] Receipt, rating, trace, and exported-candidate retention and deletion behavior are documented and tested; deletion never silently cascades across those separate stores.

## Ticket map and dependency order

```text
EGI-1  Turn-level provenance receipts
|-- EGI-2  Explicit useful / not-useful feedback
`-- EGI-3  Versioned code-review playbook and fingerprints
          `-- EGI-4  Frozen trace-to-benchmark export
EGI-2 + EGI-4
`-- EGI-5  Human-curated feedback-to-eval queue
EGI-3 + EGI-5
`-- EGI-6  Paired instruction evaluation and promotion gate
EGI-6
`-- EGI-7  Manual evidence-backed patch proposer
```

EGI-2 and EGI-3 may proceed in parallel after EGI-1. Everything after them is deliberately ordered so the project cannot build an autonomous improver before it has traceability, curation, and a regression gate.

Tickets:

- [ ] EGI-1: #488 turn-level feedback provenance receipts
- [ ] EGI-2: #489 explicit useful / not-useful feedback in the REPL
- [ ] EGI-3: #490 versioned code-review playbook with fingerprint
- [ ] EGI-4: #491 frozen replay provenance and trace-to-llm-bench export
- [ ] EGI-5: #492 human-curated feedback-to-eval queue
- [ ] EGI-6: #493 paired playbook evaluation and promotion gate
- [ ] EGI-7: #494 manual evidence-backed playbook patch proposer

## Non-goals

- Model fine-tuning, online learning, or weight updates.
- Treating implicit file-open behavior as answer-quality ground truth.
- Automatically converting every negative rating into a benchmark.
- Automatically editing, committing, pushing, merging, or deploying a prompt.
- A generic skill marketplace, registry, DSL, or workflow engine.
- Multiple self-improvement loops in the first release.
- A built-in Golem scheduler or daemon.
- Using memories as procedural policy.
- Wiring AgentFlow planning/task modes into feedback or trace capture.

## Deferred follow-up: recurring runs

After several manual cycles show that the gate is stable, review effort is reasonable, and promoted changes improve held-out results, a separate ticket may add a scheduled GitHub Action. It should produce a proposed patch and evidence report only; it should never merge or deploy automatically.
